consensus/src/main/java/org/ethereum/beacon/consensus/spec/GenesisFunction.java

```diff
@@ -55,18 +55,14 @@ default BeaconState initialize_beacon_state_from_eth1(
     state.setLatestBlockHeader(get_block_header(get_empty_block()));
 
     // Process deposits
-    for (int index = 0; index < deposits.size(); index++) {
-      Deposit deposit = deposits.get(index);
-      ReadList<Integer, DepositData> deposit_data_list =
-          ReadList.wrap(
-              deposits.stream().map(Deposit::getData).limit(index + 1).collect(Collectors.toList()),
-              Integer::new,
-              1L << getConstants().getDepositContractTreeDepth().getIntValue());
-      state.setEth1Data(new Eth1Data(
-          hash_tree_root(deposit_data_list), UInt64.valueOf(deposits.size()), eth1_block_hash));
-      verify_deposit(state, deposit);
-      process_deposit(state, deposit);
-    }
+    ReadList<Integer, DepositData> deposit_data_list =
+        ReadList.wrap(
+            deposits.stream().map(Deposit::getData).collect(Collectors.toList()),
+            Integer::new,
+            1L << getConstants().getDepositContractTreeDepth().getIntValue());
+    state.setEth1Data(new Eth1Data(
+        hash_tree_root(deposit_data_list), UInt64.valueOf(deposits.size()), eth1_block_hash));
+    deposits.forEach(deposit -> process_deposit(state, deposit));
```

Comment:
I'd add comment why verification is omitted

Reply:
Done!